runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -7908,42 +7908,47 @@ J9::X86::TreeEvaluator::VMarrayStoreCHKEvaluator(
       // -------------------------------------------------------------------------
       //
       // If the component type is java.lang.Object then the store always succeeds.
+      // On AOT, only do this when using the SVM; otherwise, it's safer to just
+      // skip this check than 1) incorrectly relocate the java/lang/Object class
+      // or 2) crash while trying to relocate.
       //
       // -------------------------------------------------------------------------
-
-      TR_OpaqueClassBlock *objectClass = fej9->getSystemClassFromClassName("java/lang/Object", 16);
-
-      if (TR::Compiler->target.is64Bit())
+      if (!cg->needClassAndMethodPointerRelocations() || comp->getOption(TR_UseSymbolValidationManager))
          {
-#ifdef J9VM_INTERP_COMPRESSED_OBJECT_HEADER
-         TR_ASSERT((((uintptr_t)objectClass) >> 32) == 0, "TR_OpaqueClassBlock must fit on 32 bits when using class pointer compression");
-         instr = generateRegImmInstruction(CMP4RegImm4, node, destComponentClassReg, (uint32_t) ((uint64_t) objectClass), cg);
+         TR_OpaqueClassBlock *objectClass = fej9->getSystemClassFromClassName("java/lang/Object", 16);
+         objectClass = fej9->getArrayClassFromComponentClass(objectClass);
```

Comment:
Should `objectClass` (and `objectClassReg` below) perhaps be renamed now that it isn't `Object` but `Object[]`?

I realize that `destComponentClassReg` is similarly misleading at this point, but in the following code the dest array class in it will be overwritten with the component class.